src/safeposix/syscalls/fs_calls.rs

```diff
@@ -5083,59 +5083,127 @@ impl Cage {
         shmid 
     }
 
-    //------------------SHMAT SYSCALL------------------
 
+    /// ### Description
+    /// 
+    /// The `shmat_syscall` maps the shared memory segment associated with the shared memory 
+    /// identifer onto the address space of the calling Cage object. The address to which the
+    /// segment is mapped is given by the `shmaddr` parameter of this function.
+    /// 
+    /// ### Arguments
+    /// 
+    /// `shmid` : identifier of the memory segment to be mapped
+    /// `shmaddr` : Address in the address space of the calling Cage where the segment is to be mapped
+    /// `shmflag` : Flag which indicates if the memory segment to be mapped is readonly or not
+    /// 
+    /// ### Returns
+    /// 
+    /// Returns the address at which the memory segment has been mapped into
+    /// 
+    /// ### Errors 
+    /// 
+    /// * `EINVAL` : If the shmid passed as an argument is an invalid identifier
+    /// 
+    /// ### Panics
+    /// 
+    /// Currently there are no scenarios where the function panics
+    /// 
+    /// For more information - refer the documentation at [https://man7.org/linux/man-pages/man3/shmat.3p.html]
     pub fn shmat_syscall(&self, shmid: i32, shmaddr: *mut u8, shmflg: i32) -> i32 {
+        // Get shm table
         let metadata = &SHM_METADATA;
         let prot: i32;
```

Comment:
Why are we using the `prot` flag? I see we are assigning either `PROT_READ` or `PROT_WRITE` flags to this, but couldn't understand it's usecase.

Reply:
Yeah seems that is not being used anywhere

Reply:
Its being passed to the `segment.map_shm` function.